app/lanes.go

```diff
@@ -0,0 +1,118 @@
+	signerAdapter := signerextraction.NewDefaultAdapter()
+
+	// 2. Create the configurations for each lane. These configurations determine how many
+	// transactions the lane can store, the maximum block space the lane can consume, and
+	// the signer extractor used to extract the expected signers from a transaction.
+
+	// Create a mev configuration that accepts maxTxPerMEVLane transactions and consumes mevLaneBlockspacePercentage of the
+	// block space.
+	mevConfig := base.LaneConfig{
+		Logger:          app.Logger(),
+		TxEncoder:       txConfig.TxEncoder(),
+		TxDecoder:       txConfig.TxDecoder(),
+		MaxBlockSpace:   mevLaneBlockspacePercentage,
+		SignerExtractor: signerAdapter,
+		MaxTxs:          maxTxPerMEVLane,
+	}
+
+	// Create a free configuration that accepts maxTxPerFreeLane transactions and consumes freeLaneBlockspacePercentage of the
+	// block space.
+	freeConfig := base.LaneConfig{
+		Logger:          app.Logger(),
+		TxEncoder:       txConfig.TxEncoder(),
+		TxDecoder:       txConfig.TxDecoder(),
+		MaxBlockSpace:   freeLaneBlockspacePercentage,
+		SignerExtractor: signerAdapter,
+		MaxTxs:          maxTxPerFreeLane,
+	}
+
+	// Create a default configuration that accepts maxTxPerDefaultLane transactions and consumes defaultLaneBlockspacePercentage of the
+	// block space.
+	defaultConfig := base.LaneConfig{
+		Logger:          app.Logger(),
+		TxEncoder:       txConfig.TxEncoder(),
+		TxDecoder:       txConfig.TxDecoder(),
+		MaxBlockSpace:   defaultLaneBlockspacePercentage,
+		SignerExtractor: signerAdapter,
+		MaxTxs:          maxTxPerDefaultLane,
+	}
+
+	// 3. Create the match handlers for each lane. These match handlers determine whether or not
+	// a transaction belongs in the lane.
+
+	// Create the final match handler for the mev lane.
+	factory := mevlane.NewDefaultAuctionFactory(txConfig.TxDecoder(), signerAdapter)
+	mevMatchHandler := factory.MatchHandler()
+
+	// Create the final match handler for the free lane.
+	freeMatchHandler := WithdrawStakingRewardsMatchHandler()
+
+	// Create the final match handler for the default lane. I.e this will direct all txs that are
+	// not free nor mev to this lane
+	defaultMatchHandler := base.DefaultMatchHandler()
+
+	// 4. Create the lanes.
+	mevLane := mevlane.NewMEVLane(
+		mevConfig,
+		factory,
+		mevMatchHandler,
+	)
+
+	freeLane := freelane.NewFreeLane(
+		freeConfig,
+		base.DefaultTxPriority(),
+		freeMatchHandler,
+	)
+
+	defaultLane := defaultlane.NewDefaultLane(
+		defaultConfig,
+		defaultMatchHandler,
+	)
+
+	return mevLane, freeLane, defaultLane
+}
+
+// WithdrawStakingRewardsMatchHandler is a match handler that matches transactions that contain a withdraw staking rewards message
+func WithdrawStakingRewardsMatchHandler() base.MatchHandler {
+	return func(ctx sdk.Context, tx sdk.Tx) bool {
+		for _, msg := range tx.GetMsgs() {
+			if _, ok := msg.(*distrtypes.MsgWithdrawDelegatorReward); ok {
+				return true
+			}
+		}
+
+		return false
+	}
+}
```

Comment:
Note: This means that any transaction including a `MsgWithdrawDelegatorReward` will be considered free. I'm assuming most delegator reward withdraw txs only include a single message?

Reply:
This is worth addressing before this gets merged.